model/task_lifecycle.go

```diff
@@ -250,21 +251,21 @@ func activatePreviousTask(ctx context.Context, taskId, caller string, originalSt
 		return activatePreviousTask(ctx, t.GeneratedBy, caller, t)
 	}
 
-	// if this is the first time we're running the task, or it's finished, has a negative priority, or already activated
-	if prevTask == nil || prevTask.IsFinished() || prevTask.Priority < 0 || prevTask.Activated {
```

Comment:
I reversed this logic and only made the `SetActiveState` rely on it. I noticed a small bug/race condition in which if two generated tasks fail, the 1st one will activate and call `SetGeneratedTasksToActivate`, then the second one will reach this condition and then `return nil`, never calling `SetGeneratedTasksToActivate`.